src/app/todo/page.tsx

```diff
@@ -0,0 +1,64 @@
+"use client";
+
+import { useRedirect } from "@/hooks/useRedirect";
+import { useUser } from "@/hooks/useUser";
+import { useState } from "react";
+import { v4 as uuid } from "uuid";
+
+export default function Todo() {
+  const [newTodo, setNewTodo] = useState("");
+  const [todos, setTodos] = useState<any[]>([]);
+  const [metaData, setMetadata] = useState({ owner: "J.Doe", count: 0 });
+
+  const user = useUser();
+
+  if (!user.loggedIn) {
+    useRedirect();
+  }
+
+  const addTodo = () => {
+    setTodos([...todos, { key: uuid(), name: newTodo, completed: false }]);
+    setNewTodo("");
+    metaData.count++;
+  };
```

Comment:
The addTodo function should not be modifying the state of the todos array directly. Instead, it should be using the setTodos function to update the state.